ql/src/java/org/apache/hadoop/hive/ql/stats/StatsUtils.java

```diff
@@ -454,6 +468,27 @@ private static Statistics collectStatistics(HiveConf conf, PrunedPartitionList p
     return stats;
   }
 
+  private static Map<String, Long> getValidBasicStatsFromStorageHandler(Table table) {
+
+    Map<String, String> storageHandlerBasicStatsMap =
+        (table.isNonNative() && table.getStorageHandler().canProvideBasicStatistics()) ? table.getStorageHandler()
+            .getBasicStatistics(Partish.buildFor(table)) : null;
```

Comment:
How about moving this check and fetch into `BasicStats` and `StatsUtils.collectStatistics` don't have don't have to be changed at all?
Example: https://github.com/kasakrisz/hive/commit/e03441477a0c616f1b3012dc4ddc6dfe0f01743b

Reply:
Done.

Reply:
Done.